python/lsst/ip/isr/photodiode.py

```diff
@@ -0,0 +1,260 @@
+            self.updateMetadata(day_obs=kwargs['day_obs'])
+        if 'seq_num' in kwargs:
+            self.updateMetadata(seq_num=kwargs['seq_num'])
+
+        self.requiredAttributes.update(['timeSamples', 'currentSamples', 'integrationMethod'])
+
+    @classmethod
+    def fromDict(cls, dictionary):
+        """Construct a PhotodiodeCalib from a dictionary of properties.
+
+        Parameters
+        ----------
+        dictionary : `dict`
+            Dictionary of properties.
+
+        Returns
+        -------
+        calib : `lsst.ip.isr.PhotodiodeCalib`
+            Constructed photodiode data.
+
+        Raises
+        ------
+            Raised if the supplied dictionary is for a different
+            calibration type.
+        """
+        calib = cls()
+
+        if calib._OBSTYPE != dictionary['metadata']['OBSTYPE']:
+            raise RuntimeError(f"Incorrect photodiode supplied.  Expected {calib._OBSTYPE}, "
+                               f"found {dictionary['metadata']['OBSTYPE']}")
+
+        calib.setMetadata(dictionary['metadata'])
+
+        calib.timeSamples = np.array(dictionary.get('timeSamples'))
+        calib.currentSamples = np.array(dictionary.get('currentSamples'))
+        calib.integrationMethod = dictionary.get('integrationMethod')
+
+        calib.updateMetadata()
+        return calib
+
+    def toDict(self):
+        """Return a dictionary containing the photodiode properties.
+
+        The dictionary should be able to be round-tripped through.
+        `fromDict`.
+
+        Returns
+        -------
+        dictionary : `dict`
+            Dictionary of properties.
+        """
+        self.updateMetadata()
+
+        outDict = {}
+        metadata = self.getMetadata()
+        outDict['metadata'] = metadata
+
+        outDict['timeSamples'] = self.timeSamples.tolist()
+        outDict['currentSamples'] = self.currentSamples.tolist()
+
+        outDict['integrationMethod'] = self.integrationMethod
+
+        return outDict
+
+    @classmethod
+    def fromTable(cls, tableList):
+        """Construct calibration from a list of tables.
+
+        This method uses the `fromDict` method to create the
+        calibration after constructing an appropriate dictionary from
+        the input tables.
+
+        Parameters
+        ----------
+        tableList : `list` [`astropy.table.Table`]
+            List of tables to use to construct the crosstalk
+            calibration.
+
+        Returns
+        -------
+        calib : `lsst.ip.isr.PhotodiodeCalib`
+            The calibration defined in the tables.
+        """
+        dataTable = tableList[0]
+
+        metadata = dataTable.meta
+        inDict = {}
+        inDict['metadata'] = metadata
+        inDict['integrationMethod'] = metadata.pop('INTEGRATION_METHOD', None)
+
+        inDict['timeSamples'] = dataTable['TIME']
+        inDict['currentSamples'] = dataTable['CURRENT']
+
+        return cls().fromDict(inDict)
```

Comment:
Ah, it looks like that's exactly what it is. Arguably I should read everything through, take notes, and then start writing review comments, but that's like 10x the work, so instead I'm just sorry I've not done that 😔